src/main/java/org/jabref/gui/fieldeditors/MapBasedEditorViewModel.java

```diff
@@ -38,7 +39,11 @@ public T fromString(String string) {
                 if (string == null) {
                     return null;
                 } else {
-                    return getItemMap().getOrDefault(string, null);
+
+                    if (getItemMap().get(string) == null) {
+                        getItemMap().put(string, (T) string);
```

Comment:
Ok thanks!
According to the documentation
> a string converter property is provided to allow for developers to specify how to translate a users string into an object of type T, such that the value property may contain it. By default the converter simply returns the String input as the user typed it, which therefore assumes that the type of the editable ComboBox is String.

So you are right that the string converter should return something non-null even when the text is not recognized. However, there is no need to add it to the item map. Moreover, we need to handle the case when `T` is not a string type. Thus I would propose to create a new protected method that converts an unknown string to a `T` object. If not overridden, it should try to cast to `T` as in your code above (and throw an exception with details how to properly implement it if casting was not successful).

Reply:
I now added a method for checking.
We also need such kind for dates to allow custom values which we currently don't support. All input in the field will be cleared if not parseable